src/reactA11yInputElementsRule.ts

```diff
@@ -61,6 +61,21 @@ function isExcludedInputType(node: ts.JsxSelfClosingElement, attributes: { [prop
     return false;
 }
 
+function isInputTypeFile(node: ts.JsxSelfClosingElement, attributes: { [propName: string]: ts.JsxAttribute }): boolean {
+    for (const attribute of node.attributes.properties) {
+        if (tsutils.isJsxAttribute(attribute)) {
+            const isInputAttributeType = attributes.type;
```

Comment:
> `isInputAttributeType`

This is a bit of an existing concern, but the inner two `if` statements are not necessary if `isInputAttributeType` is falsy. Could you change these functions to return before the `for` loop if that's the case? Seems like a lot of wasted effort here.

Reply:
Oh, I couldn't find it. These codes are a little complex to read.